Show active exception or interrupt in thread description

kjbracey-arm opened the pull request:
Implementation is straightforward, just not totally certain where each bit should live in the structure.

kjbracey-arm commented:
IRQ names would make it even more awesome. So, do you want me to incorporate that commit into this PR now?

kjbracey-arm commented:
Added your commit as-is, then a fixup commit combining the two.

Now shows CMSIS SVD interrupt name.

flit commented on pyocd/coresight/cortex_m.py:
```diff
@@ -937,3 +894,39 @@ def get_target_context(self, core=None):
+           "MemManage",
+           "BusFault",
+           "UsageFault",
+           "Exception 7",
```
Should go ahead and add SecureFault.

flit commented on pyocd/coresight/cortex_m.py:
```diff
@@ -937,3 +894,39 @@ def get_target_context(self, core=None):
+                return self.CORE_EXCEPTION[num]
+        else:
+            irq_num = exc_num - len(self.CORE_EXCEPTION)
+            name = None;
```
Semicolon 😉

flit commented on pyocd/gdbserver/gdbserver.py:
```diff
@@ -1309,6 +1309,13 @@ def is_threading_enabled(self):
 
+    def exception_name(self):
+        try:
+            ipsr = self.target_context.read_core_register('xpsr') & 0x1ff
```
I'd like to get #436 in first, then make use of `get_ipsr()` here.

flit commented:
Ok, now that #436 is merged in, if you update `GDBServer.exception_name()`, I'll get this one in, too. Thanks!

kjbracey-arm commented:
Rebased and adjusted. Are you happy with the `Interrupt[%s]` and `Interrupt %d` formatting?

Not 100% sure whether I like those, but not sure what would be better.

kjbracey-arm commented:
> The only thing I can think of that might improve the descriptions is to always include the irq number, something like `Interrupt 17` and `Interrupt 29 (UART1)`. The irq number is important if you are correlating with NVIC registers.

I think this (just about) comes under my "clutter" category, but it is marginal. The actual interrupt number is still available if you happen to need it, by examining IPSR. Don't need to show all metadata in the thread list.

Guess it could be `Interrupt 29 [UART1]` though. 

I nicked the current style from the RTX `Waiting[xxx]`, and Eclipse also uses `[]` for thread names - although GDB uses quotes. Want to avoid round brackets as description is shown in another set of round brackets.
